### PR TITLE
Puppet debug/parser options

### PR DESCRIPTION
**Pull request by carlweise-msm**
* Added ability to control puppet parser and debug via inputs.

### CODE REVIEW

File: init.sh [hunk 1]
```diff
@@ -111,7 +113,12 @@ parse_args() {
         SECURE_PUPPET="${2}"
         shift
         ;;
+      --parser)
+        PUPPET_PARSER="${$2}"
```
Comment (andy-smith-msm) [inline init.sh]: ${$2} ?

Comment (carlweise-msm) [reply init.sh]: Thanks, fixed!

File: init.sh [hunk 2]
```diff
@@ -519,7 +530,7 @@ run_puppet() {
   echo ""
   echo "Running puppet apply"
   export FACTERLIB="${FACTERLIB}:$(ipaddress_primary_path)"
-  puppet apply /etc/puppet/manifests/site.pp --detailed-exitcodes
+  puppet apply ${PUPPET_DEBUG_OPT} ${PUPPET_PARSER_OPT} /etc/puppet/manifests/site.pp --detailed-exitcodes
```
Comment (andy-smith-msm) [inline init.sh]: Any reason not to make this simple PUPPET_APPLY_OPTS and use one generic value?

Comment (andy-smith-msm) [reply init.sh]: (one would simply quote the multi values to get it into the single arg)

Comment (carlweise-msm) [reply init.sh]: None at all, if you think this would be easier to read I am happy to make the change.

**Pull request merged**